test/integration/websocket_integration_test.cc

```diff
@@ -237,27 +237,20 @@ TEST_P(WebsocketIntegrationTest, EarlyData) {
 }
 
 TEST_P(WebsocketIntegrationTest, WebSocketConnectionIdleTimeout) {
-  if (!old_style_websockets_) {
-    return;
-  }
-
   envoy::api::v2::route::RouteAction::WebSocketProxyConfig ws_config;
   ws_config.mutable_idle_timeout()->set_nanos(
       std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::milliseconds(100)).count());
   *ws_config.mutable_stat_prefix() = "my-stat-prefix";
   config_helper_.addConfigModifier(setRouteUsingWebsocket(&ws_config, old_style_websockets_));
   if (!old_style_websockets_) {
     config_helper_.addConfigModifier(
-        [](envoy::config::filter::network::http_connection_manager::v2::HttpConnectionManager&
-               hcm) {
-          auto* route = hcm.mutable_route_config()->mutable_virtual_hosts(0)->mutable_routes(0);
-          auto* route_idle_timeout = route->mutable_route()->mutable_timeout();
-          auto nanos =
-              std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::milliseconds(100))
-                  .count();
-          route_idle_timeout->set_nanos(nanos);
-          auto* idle_timeout = hcm.mutable_idle_timeout();
-          idle_timeout->set_nanos(nanos);
+        [&](envoy::config::filter::network::http_connection_manager::v2::HttpConnectionManager& hcm)
+            -> void {
+          auto* route_config = hcm.mutable_route_config();
```

Comment:
This change seems unrelated to deprecating this. Should it be in a separate PR?

Reply:
If you'd prefer I'd be happy to.  The thing holding me back from deprecation was the lack of idle timeouts, so to me it's finishing the feature and deprecating the old version.  If we think it's worth splitting out it's easy enough on my end.